### ISSUE TITLE
Controls on UserForms should resolve the AsTypeName to a coclass instead of an interface.

### ISSUE THREAD

**Issue by comintern** [rubberduck-vba/Rubberduck#2593]
Currently, RD is querying the `IDispatch` interface for controls on UserForms and using the default interface for the type of the `Declaration`:

![untitled](https://cloud.githubusercontent.com/assets/8944005/22407434/cb324f40-e62b-11e6-8b3d-b1de77f8ac14.png)

While this is _technically_ correct from a purely COM perspective and allows for member resolution, VBA treats them as instances of the coclass, and it would probably be more meaningful to users to treat them the same way:

```
Private Sub UserForm_Activate()
    Debug.Print TypeName(Label1)  '<--Prints "Label"
End Sub
```
In fact in fm20.dll at least, the default interfaces for the coclasses they're associated with are hidden (that's why you won't find `ILabelControl` in the Object Browser).  

RD should resolve these as coclasses, probably via a default interface->coclass  lookup table.

**Comment by comintern** [comment]
Linking #2592.

**Comment by ThunderFrame** [comment]
As you'd expect:

```vb
Private Sub UserForm_Activate()
  Debug.Print TypeName(Me.Label1)                       'Prints Label
  Debug.Print TypeName(Me.Label1.Object)                'Prints ILabelControl
  Debug.Print TypeName(Me.Controls(Label1.Name))        'Prints Label
  Debug.Print TypeName(Me.Controls(Label1.Name).Object) 'Prints ILabelControl
  Debug.Print TypeName(Me.Controls("Label1").Object)    'Prints ILabelControl
End Sub
```

**Comment by ThunderFrame** [comment]
Those extension members are in there too:

```vb
  Me.MultiPage1.Page1.Caption = "FOO" 'Page1 is an extension member,not in Intellisense or OB
  Me.TabStrip1.Tab1.Caption = "FOO"   'Tab1 is an extension member,not in Intellisense or OB
```

**Comment by ThunderFrame** [comment]
I *think* I remember reading somewhere that Tabs and Pages are essentially child forms.

**Comment by comintern** [comment]
@ThunderFrame - they're container controls. `Tabs` and `Pages` are the default properties of `ITabStrip` and `IMultipage`, so it appears to be using an extended interface and passing them on to the default.

```
//ITabStrip
[id(00000000), propget, helpcontext(0x001e8c78)]
Tabs* Tabs();
```

```
//IMultiPage:
[id(00000000), propget, helpcontext(0x001e8b06)]
Pages* Pages();
```
I'd say any inspection results for `Me.MultiPage1.Page1.Caption` etc. would be legit - this would be preferable syntax for a one-liner:

`Me.MultiPage1.Pages("Page1").Caption = "Foo"  '<-- This is probably better.`

Better would be to use an intermediate `Page` object.

**Comment by ThunderFrame** [comment]
Ditto for `Frames`

```vb
Debug.Print UserForm1.Frame1.OptionButton1.Caption
```

**Comment by ThunderFrame** [comment]
@comintern yep, seems the default property is accessible with a bang/shriek too. Afterall, it's just passing the membername as a string argument to the default member.

Both of these work:
```vb
Debug.Print UserForm1.Frame1!OptionButton1.Caption
Debug.Print UserForm1!Frame1!OptionButton1.Caption
```

**Comment by comintern** [comment]
Closed via f1a5df6
